user/src/com/google/gwt/user/server/rpc/EnumMapUtil.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2022 Google Inc.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ * 
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.gwt.user.server.rpc;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.ObjectStreamClass;
+
+/**
+ * A utility to extract the key type of any {@link java.util.EnumMap}, even if it is empty. Some
+ * frameworks have tried this by using reflection on the map's private field {@code keyType}, but
+ * that approach breaks with Java 17 where reflection on private elements of any {@code java.*}
+ * class is forbidden unless the {@code --add-opens} VM argument is used which is a fairly unsafe
+ * thing to do.
+ * <p>
+ * 
+ * Use like this: <pre>
+ *      private static enum MyBoolean {
+ *          TRUE, FALSE
+ *      };
+ *      EnumMap<MyBoolean, Integer> enumMap = new EnumMap<>(MyBoolean.class);
+ *      Class<?> c = EnumMapUtil.getKeyType(enumMap);
+ * </pre>
+ * <p>
+ * 
+ * Implementation note: If the map passed to {@link #getKeyType(java.util.EnumMap)} is not empty,
+ * the first key is obtained from the {@link EnumMap#keySet() key set} and its type is determined
+ * and returned. If the map is empty, it is serialized into an {@link ObjectOutputStream} writing to
+ * a {@link ByteArrayOutputStream} from which it is read again with a specialized
+ * {@link ObjectInputStream}where the {@code resolveClass} method is overridden. Upon reading a
+ * class descriptor for class {@link java.util.EnumMap}, instead of regularly resolving the class
+ * descriptor a package-protected class {@link com.google.gwt.user.server.rpc.EnumMap} is returned which has a
```

Comment:
Please check the diff and wrap the long lines in this file

Reply:
Same as above. Fixed with a838a02fbfa96c2b806e57005d3667bb7763ca4d.